### DISCUSSION TITLE
Query regarding affiliation with Azure.

### DISCUSSION THREAD

**Discussion by syfai** [rany2/edge-tts#261] [General]
Hey There!
I'm just concerned if this repository (Edge-TTS) is anyway affiliated with Azure Cognitive Services' Text-to-Speech or any of the Microsoft's proprietary services? Like, for speech generations 
### **is this relying on the Azure AI Speech?** 
If so, it can't be used for commercial purposes. 
Please can anybody from the official team of this repository answer my question, as it is crucial for usage purposes. 
And also I hope, it is working on its own, not affiliated to Microsoft.
You know, I was just wondering if I can make a TTS website, and commercialize this with generous subscription plans or ads, also will be following the required conditions.

**Comment by WillJDunn** [comment]
If you review the code you can see the specific service which is being used: https://github.com/rany2/edge-tts/blob/master/src/edge_tts/constants.py
```Python
WSS_URL = (
    "wss://speech.platform.bing.com/consumer/speech/synthesize/"
    + "readaloud/edge/v1?TrustedClientToken="
    + TRUSTED_CLIENT_TOKEN
)
```

In fact it appears that this is the Bing Speech Platform. If you [demo the voices available in the Azure Speech API](https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/new-hd-voices-preview-in-azure-ai-speech-contextual-and/ba-p/4258325) you'll find they are the same as the ones available in edge-tts (for example, compare edge-tts' "en-US-SteffanNeural" voice to the Azure API's "en-US-Steffan")

As such I think it's safe to conclude that you can use edge-tts for your personal use and the Azure API for commercial use. Unless others have additional information - I'm not seeing licensing information for the bing speech platform anymore.

**Reply by syfai** [reply]
Oh hi Will,
Thanks for your reply. I got you, and thanks for your explanation. 

Just my views after reading your comment:
If that's the case then why does the license is under the GNU General Public License v3.0? This license gives right to commercialize.
Isn't this misleading? The repository holders should need to change it immediately that its not for commercial use. As commercial use without a valid Azure subscription is a violation of Microsoft's terms of service. And if somebody commercialized by just reading the license given here, then he could be in trouble.
I asked MS support community as well and got similar reply
Please check out: https://learn.microsoft.com/en-us/answers/questions/2088770/are-opensource-edge-tts-free-for-commercial-use

**Reply by rany2** [reply]
It's not really misleading or unusual. For example alternative YouTube clients like Invidious are licensed under AGPLv3 which is even more restrictive than GPLv3. The license allows you to use it commercially at your own risk.

I might add a disclaimer against using it for commercial reasons. This library is meant for personal use. I've only used edge-tts for Home Assistant and to convert my personal documents into ebooks.

It's generally not a good idea to use this library commercially for two reasons, one selfish and the other more ethical:

1. The selfish reason is that this library is absolutely not reliable and could stop working at any moment and without warning. In addition, you're not subject to any guarantees about availability or rate limits. Basically you shouldn't have your business rely on something this risky. Granted it's been working fine for more than 5 years now, but I'd still never risk it for commercial use.
2. The ethical reason is that you're making profit from a service that isn't cheap to replicate; at least not at this quality and with this variety of voices. I'm not suggesting that they'll shut down the service if you don't use it via their API, but it probably doesn't help them improve the service further either.

**Reply by syfai** [reply]
Dear @rany2, I'm thankful for the clarification you provided.

**Reply by WillJDunn** [reply]
> Just my views after reading your comment: If that's the case then why does the license is under the GNU General Public License v3.0? This license gives right to commercialize. Isn't this misleading? The repository holders should need to change it immediately that its not for commercial use. As commercial use without a valid Azure subscription is a violation of Microsoft's terms of service. And if somebody commercialized by just reading the license given here, then he could be in trouble. I asked MS support community as well and got similar reply Please check out: https://learn.microsoft.com/en-us/answers/questions/2088770/are-opensource-edge-tts-free-for-commercial-use

Regarding the use of GPL V3 and similar licenses, if I'm not misunderstanding it is just regarding the copyright of the dev's source code, and explicitly states that the code is shared without WARRANTY or LIABILITY. We should not think of it as providing any promises or guarantees about the code's performance, compliance with your region's laws, compliance with any 3rd party terms and conditions (such as Microsoft's), etc.. When the GPL V3 and similar licenses do not restrict you from using their code commercially it is only a lack of restriction from the author's copyright and not any promise of commercial viability. So GPL V3 is always an acceptable license even for open source projects which cannot be commercialized for various reasons. However, information regarding such issues are often provided as additional disclaimers and T&Cs - might be a helpful contribution.
